src/NuGet.Clients/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -108,11 +108,13 @@ public override async Task ExecuteCommandAsync()
                 var v2RestoreResult = await PerformNuGetV2RestoreAsync(restoreInputs);
                 restoreSummaries.Add(v2RestoreResult);
 
-                // log warnings
-                v2RestoreResult
-                    .Errors
-                    .Where(l => l.Level == LogLevel.Warning)
-                    .ForEach(l => Console.LogWarning(l.FormatWithCode()));
+                if (!v2RestoreResult.Success)
```

Comment:
Could you have warnings even when the result is successful?

Reply:
That is now logged [here](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/PackageExtractor.cs#L1017) since we are now passing a valid logger all the way down to the extractor.

Reply:
but that's only if the package is signed, are there any warnings for unsigned packages that we should be logging here?

Reply:
discussed offline. The problem only happens in the case of a signature failure during extraction. The extractor simply [throws](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/PackageExtractor.cs#L1029) an exception, hence if the restore fails, we need to log the warnings post restore.